src/components/RegistrationForm.tsx

```diff
@@ -0,0 +1,89 @@
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+/* eslint-disable jsx-a11y/anchor-is-valid */
+/* eslint-disable max-len */
+
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import emailIcon from '../assests/icons/email.png';
+import passwordIcon from '../assests/icons/password.png';
+import { useLocalStorage } from '../utils/hooks/useLocalStorage';
+
+type Props = {
+  setShowForm: (boolean: boolean) => void,
+};
+
+export const RegistrationForm:React.FC<Props> = ({ setShowForm }) => {
+  const [data, setData] = useLocalStorage('userData', {
+    email: '',
+    password: '',
+  });
+
+  const [user, setUser] = useState({
+    email: '',
+    password: '',
+  });
+  const [touched, setTouched] = useState(false);
+  const EMAIL_REGEXP = /^(([^<>()[\].,;:\s@"]+(\.[^<>()[\].,;:\s@"]+)*)|(".+"))@(([^<>()[\].,;:\s@"]+\.)+[^<>()[\].,;:\s@"]{2,})$/iu;
+
+  const handleAddUser = () => {
+    setTouched(true);
+    if (EMAIL_REGEXP.test(user.email) && user.password.length) {
+      setData({ ...data, ...user });
+      setShowForm(false);
+    }
+  };
+
+  return (
+    <>
+      <div className="login-page">
+        <h1 className="title">Login</h1>
+        <div
+          className="form"
+          onKeyDown={(e) => {
+            if (e.key === 'Enter') {
+              handleAddUser();
+            }
+          }}
+        >
+          <div className={classNames(
+            'form__item',
+            { 'is-danger': touched && !EMAIL_REGEXP.test(user.email) },
+          )}
+          >
+            <input
+              type="email"
+              placeholder="Enter your email"
+              required
+              defaultValue={user.email}
+              onChange={(e) => setUser({ ...user, email: e.target.value })}
+            />
+            <img src={emailIcon} alt="" width={28} />
```

Comment:
It is a good practice to provide a meaningful value for the `alt` attribute on images for accessibility reasons.